TravelSchedule/View/AgreementView/WebView.swift

```diff
@@ -0,0 +1,52 @@
+import SwiftUI
+import WebKit
+
+struct WebView: UIViewRepresentable {
+    let urlString: String
+    @ObservedObject var viewModel: ProgressViewModel
+    private let webView = WKWebView()
+    
+    func makeUIView(context: Context) -> WKWebView {
+        if let url = URL(string: urlString) {
+            webView.load(URLRequest(url: url))
+        }
+        return webView
+    }
+    
+    func updateUIView(_ uiView: WKWebView, context: Context) {}
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(parent: self, viewModel: viewModel)
+    }
+    
+    class Coordinator: NSObject, WKNavigationDelegate {
+        private var parent: WebView
+        private var viewModel: ProgressViewModel
+        private var observer: NSKeyValueObservation?
+        
+        init(parent: WebView, viewModel: ProgressViewModel) {
+            self.parent = parent
+            self.viewModel = viewModel
+            super.init()
+            
+            self.observer = self.parent.webView.observe(\.estimatedProgress) { [weak self] webView, _ in
+                guard let self else { return }
+                self.parent.viewModel.progress = parent.webView.estimatedProgress
```

Comment:
Проблема `Publishing changes from within view updates is not allowed, this will cause undefined behavior` возникает из-за того, что сейчас `WebView` обновляет `@Published`-свойство `progress`.

Должен помочь перенос операции изменения значения в главную очередь:
```
DispatchQueue.main.async { 
                    self.parent.viewModel.progress = parent.webView.estimatedProgress
                }
```

Reply:
Да, на stackOverFlow тоже нашел ответ, что да, нужно перенести в главную очередь. Попробую.